Blocks/Persistence/persistence.h

```diff
@@ -323,14 +375,24 @@ struct CerealAppendToFilePublisherImpl {
   explicit CerealAppendToFilePublisherImpl(const std::string& filename) : filename_(filename) {}
 
   void Replay(std::function<void(IDX_TS, ENTRY&&)> push) {
+    using current::strings::Printf;
     assert(!appender_);
     std::ifstream fi(filename_);
+    IDX_TS last_idx_ts;
     if (fi.good()) {
       IDX_TS idx_ts;
       while (fi >> idx_ts.index) {
+        if (idx_ts.index != last_idx_ts.index + 1) {
+          CURRENT_THROW(InconsistentIndexDuringReplayException(
+              Printf("Expected '%llu', found '%llu'.", last_idx_ts.index + 1, idx_ts.index)));
+        }
         int64_t timestamp;
         fi >> timestamp;
         idx_ts.us = std::chrono::microseconds(timestamp);
+        if (idx_ts.us <= last_idx_ts.us) {
```

Comment:
Good question :)
1) "Under ordinary conditions, ntpd adjusts the clock in small steps so that the timescale is effectively continuous and without discontinuities."
2) Our `time::Now()` is strictly increasing.
3) DO NOT mess with time on your server :)